### PR TITLE
Snowflake test schema builder

### PR DESCRIPTION
**Pull request by Ismail-Naziz**
## What
test if schema builder function and documentation of behavior

## How
adding new file for test

## Review guide
airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py

## User Impact
User can now understand how data is formatted and processed

## Can this PR be safely reverted and rolled back?
YES 💚

### CODE REVIEW

File: airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py
Comment (maxi297) [inline airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py:35]: Can we have those tests be parametrized or split into multiple test? I fear that if one fails, we won't get visibility on all the others

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py
Comment (maxi297) [inline airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py:159]: Can we update the comment `# maybe add warning` in the `except` statement? If I understand correctly, this is a date that would be empty, right? Why do we have this added condition? Is this a case we've seen? I would assume that since all our fields are nullable, we can return `None` in this case but if we don't expect this to happen, I would indeed add a log

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py
Comment (maxi297) [inline airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py:199]: Regarding the int conversion (at the time of this comment, it is line 204), why do we parse as a string before parsing as an int?

File: airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py
Comment (maxi297) [inline airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py:126]: nit: no need to change that but for the newly added tests, we can rely on the assert statement as this is printed way better when running with the pytest runner than `self.assertEqual`

**Review by maxi297 — APPROVED** [review]
LGTM conditional to fixing the comments

File: airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py
Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py:35]: I don't understand what you mean by parametrized. I defined the test as unit that verifies that the whole function is working not that a fraction of it is working. So, if something breaks early, you will  have fix the first issue as the test will stop at that point. Then when you rerun the test you will get next errors if there are errors.

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py
Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py]: removed

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py
Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py]: Corrected in the commit a84095b35c49a25d030b03862068c3c697774101

File: airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py
Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py]: Done

File: airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py
Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py]: Replaced in these tests

**Review by maxi297 — COMMENTED** [review]
Two small comments before I approve

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py
Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py:159]: Normally, in snowflake response. With any possible snowflake type, it is not expected to have a value that is not a string in snowflake response. I will raise a Type error with details about the input. Btw, the error will be caught as we make sure that an error with one record will never break read_records

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py
Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py:161]: I will check that at the start with an early return. first check if not string type with type error. If empty string, return empty value.

**Comment by Ismail-Naziz** [comment]
All the comments are adjusted in this PR, can you approve this one ? @maxi297

**Pull request merged**
